### DISCUSSION TITLE
How does registryUrls option work?

### DISCUSSION THREAD

**Discussion by marverix** [renovatebot/renovate#32221] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitHub

### Please tell us more about your question or problem

Hello,

I'm trying to understand how `registryUrls` option work?

I'm running self-hosted version of the newest v38. Running Renovate against multiple Java repositories. In each repository the package manager is Gradle.
In each repo I have settings.gradle file that defines the registries for (Gradle) plugins and (Maven) dependencies.

We are using Renovate local cache (`cacheDir` option).

For long time we had configured Nexus OSS as a proxy. There was a group `/maven/` which on the first place proxied to Maven Central, and then to self-hosted repository. But after some time we discovered that Nexus is overwhelmed by this configuration, so we decided to remove from `/maven/` group Maven Central position, so it's not anymore proxing it. Now `mavenCentral()` is added manually to each settings.gradle as the first position, so in theory it should be the first go-to repository.
But after that change we discovered that nothing changed. It looks like Renovate is till pinging `/maven/` instead `mavenCentral()`.

So in `default.json` I have added some rules to set `registryUrls` depending on the package name (if the package starts from our company domain or not).

But I'm wondering will it work?
How does registyUrls work? Like in deep dive. Is this array merged or overwritten each time? Is it documented somewhere? Is registryUrls cached?

Thanks!
Br,
Marek

**Comment by rarkins** [comment]
The docs for registryUrls are at https://docs.renovatebot.com/configuration-options/#registryurls

When using open source software I advise:
- Most projects don't have the time for you to ask "give me a deep dive about everything, because I'm curious"
- Don't write anything like "I added some rules - WHICH I HAVEN'T BOTHERED TO SHOW YOU HERE - will they work?
- Avoid theoretical questions like "will it work?" in general. The best is for you to test it. Do you have any reason to think it _won't_ work?
